### PR TITLE
call can_create to get the reason

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
resolves taipy-enterprise#362

### CODE REVIEW

**Review by trgiangdo — DISMISSED** [review]
LGTM

**Review by namnguyen20999 — DISMISSED** [review]

**Review by namnguyen20999 — DISMISSED** [review]
lgtm

File: taipy/gui_core/_context.py [hunk 1]
```diff
@@ -1104,3 +1105,8 @@ def on_dag_select(self, state: State, id: str, payload: t.Dict[str, str]):
                     _warn(f"dag.on_action(): Exception raised in '{args[1]}()' with '{args[0]}'", e)
         elif args[1]:
             _warn(f"dag.on_action(): Invalid function '{args[1]}()'.")
+
+    def get_creation_reason(self):
+        if not (reason := can_create()):
```
Comment (FabienLelaquais) [inline taipy/gui_core/_context.py]: def get_creation_reason(self):
```suggestion
        f"Cannot create scenario: {reason.reasons}" if (reason := can_create()) else ""
```
?

**Review by FabienLelaquais — DISMISSED** [review]
As long as `get_creation_reason()` is invoked at the right time... looks just fine!

File: taipy/gui_core/_context.py (hunk 1 above)
Comment (FredLL-Avaiga) [reply taipy/gui_core/_context.py]: it would be f"Cannot ..." if not (reason :! ...)
I suppose it's more pythonic ...

**Review by FabienLelaquais — APPROVED** [review]
Even better!

**Review by FabienLelaquais — APPROVED** [review]
🫤

**Pull request merged**